tests/integration/records/unload-test.js

```diff
@@ -419,3 +419,198 @@ test('unloading a disconnected subgraph clears the relevant internal models', fu
 
   assert.equal(relPayloads.get('person', 1, 'cars'), null, 'person - cars relationship payload unloaded');
 });
+
+
+test("Unloading a record twice only schedules destroy once", function(assert) {
+  const store = env.store;
+  let record;
+
+  // populate initial record
+  run(function() {
+    record = store.push({
+      data: {
+        type: 'person',
+        id: '1',
+        attributes: {
+          name: 'Adam Sunderland'
+        }
+      }
+    });
+  });
+
+  const internalModel = record._internalModel;
+  assert.equal(internalModel.currentState.stateName, 'root.loaded.saved', 'We are loaded initially');
+
+  run(function() {
+    let timer = null;
+    store.unloadRecord(record);
+    timer = internalModel._scheduledDestroy;
+    assert.notEqual(timer, null, 'We scheduled destroy');
+    store.unloadRecord(record);
+    let  newTimer = internalModel._scheduledDestroy;
+    assert.equal(newTimer, timer, 'We scheduled destroy only once');
```

Comment:
is there a way for us to test the observable behavior of this rather then the implementation? Specifically, i suspect this will cause issues \w resurrection, and it should be possible to test that directly.